### PR TITLE
End-to-end working localization

### PR DESCRIPTION
**Pull request by TomNUSDS**
## Creating a draft PR as a starting point to discuss odd things with Nat.

- Just started with the index page since it's pretty static (not actively being worked on).
- Some files probably should be removed from the build since they are "built" files (`compiled-lang/`) this is so we can discuss the build process.

Oddities:
- Current approach is breaking one of the intl rules of not manually assigning message ids. It is intentional and one of the topics to discuss.
  
 - Topmost template needs to wrap all the code in a `<IntlProvider>` so that `<FormattedMessage>` can be used. Interestingly, `intl.formatMessage()` does not have this requirement.
  
 -  `<IntlProvider>` requires a messages parameter passed to it. The messages need to be a key:value array where both key and value are strings... this for the current language. I've done this by directly including the es.json that the translators work on, but this seems wrong to me. 
  
 -  The `/intl/en.json` and `/intl/es.json` are different formats? Seems odd, but the workflow is that you:
 1. run`intl:extract` that generates the `en.json`... _This_ json is formated like`key:{defaultMessage, description}`
 2. this file is run through some (currently undefined) localization console tool/process generates es.json output.
 3. but that file is is `key:string` format!
 4. A "formatjs compile" is required to turn `es.json` into a _different_ format of es.json in `compiled-lang/`
 
Note: Any new strings added to the code but not found in the es.json will result in chrome logging that the key is messing when the page loads. Obviously, there needs to be some form of testing/build step to catch these.
  
 - So, `react-intl :: IntlProvider / FormattedMessage` needs `key:string` data, but `gatsby-plugin-intl :: useIntl() / intl.formatMessage()` needs it _compiled_ into `/compile-lang/es.json`! So, umm... why?

### CODE REVIEW

**Review by TomNUSDS — COMMENTED** [review]
@NatHillardUSDS Added a bunch of inline comments to give context to this **DRAFT** discussion-only PR.

File: client/compiled-lang/en.json [hunk 1]
```diff
@@ -0,0 +1,186 @@
+{
```
[the remark this answers is not in the document]
Comment (NatHillardUSDS) [reply client/compiled-lang/en.json]: Yep, according to[ this workflow](https://formatjs.io/docs/getting-started/message-extraction) we need to have some file there - we can check in an empty version to start: `{ }`

File: client/package.json [hunk 2]
```diff
@@ -25,7 +25,8 @@
     "lint": "eslint './src/**/*.{ts,tsx}' --ignore-pattern node_modules/ --ignore-pattern public --ignore-pattern *scss.d.ts",
     "lint:fix": "npm run lint -- --quiet --fix",
     "intl:extract": "formatjs extract 'src/(pages|components)/*.tsx' --out-file src/intl/en.json",
-    "intl:compile": "formatjs compile src/intl/en.json --ast --out-file compiled-lang/en.json"
+    "intl:compile-en": "formatjs compile src/intl/en.json --ast --out-file compiled-lang/en.json",
+    "intl:compile-es": "formatjs compile src/intl/es.json --ast --out-file compiled-lang/es.json"
```
[the remark this answers is not in the document]
Comment (NatHillardUSDS) [reply client/package.json:28]: yeah, it's still necessary to compile `en` to get it in the right format

**Review by NatHillardUSDS — APPROVED** [review]
Awesome! Looks great, thanks again for the conversations around this!

**Pull request merged**